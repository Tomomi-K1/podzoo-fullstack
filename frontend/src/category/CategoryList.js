import React, { useState, useEffect } from "react";
import Grid from '@mui/material/Grid';
import Container from '@mui/material/Container';

import CategoryCard from "./CategoryCard";


function CategoryList({categories}){



    return(
        // <Grid container spacing={1} >
        <Grid container spacing={1} sx={{flexGrow:1}} justifyContent="center">
            <Grid item xs={12} md={11} lg={10} sx={{ display:"flex", justifyContent:"center"}}>
                <Grid container justifyContent="center" spacing={1}>
                {categories.map(category =>{
                    return (
                    <Grid item key={category.id} xs ={3} md={1} lg={1} sx={{p:1}} >
                        <CategoryCard  category={category}/>
                    </Grid>
                    ) 
                })}
                </Grid>
            </Grid>
     </Grid>
    //     <Container sx={{display:"flex", justifyContent:"center", flexWrap:"wrap"}}>
    //     {categories.map(category =>{
    //         return (
    //            <div key={category.id}>
    //             {/* <Grid item xs ={6} md={4} lg={3} sx={{m:'auto'}} > */}
    //               <CategoryCard  category={category}/>
    //              {/* </Grid> */}
    //              </div>
    //         ) 
    //     })}
    //  {/* </Grid> */}
    //  </Container>
    )
}

export default CategoryList;