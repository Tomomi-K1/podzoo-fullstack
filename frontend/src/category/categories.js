   
const categories=[
   {id: 9, name: "Business"},
   {id: 16, name: "Comedy"},           
   {id: 20, name: "Education"},                
   // {id: 26, name: "Fiction"},      
   {id: 27, name: "Drama"},
   // {id: 28, name: "History"},
   {id: 29, name: "Health"},           
   {id: 53, name: "Music"},           
   {id: 55, name: "News"},           
   // {id: 57, name: "Entertainment"},            
   // {id: 59, name: "Politics"},            
   // {id: 67, name: "Science"},            
   // {id: 74, name: "Nature"},            
   // {id: 78, name: "Culture"},            
   // {id: 79, name: "Documentary"},            
   {id: 86, name: "Sports"},        
   {id: 102, name: "Tech"},            
   {id: 104, name: "TV"},            
   {id: 105, name: "Film"},          
   {id: 109, name: "Weather"}
]

export default categories;