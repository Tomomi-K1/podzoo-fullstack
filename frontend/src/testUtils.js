import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };