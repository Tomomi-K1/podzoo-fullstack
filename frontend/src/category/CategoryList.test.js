import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import CategoryList from "./CategoryList";
import { UserProvider } from "../testUtils";
import categories from "./categories";

it('renders', () => {
    render(
        <MemoryRouter>
            <CategoryList categories={categories}/>
        </MemoryRouter>
    ) 
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <CategoryList categories={categories}/>
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

it('has correctly renders', () => {
    render(
        <MemoryRouter>
            <CategoryList categories={categories} />
        </MemoryRouter>
        )
    expect(screen.getByText('Education')).toBeInTheDocument();
})