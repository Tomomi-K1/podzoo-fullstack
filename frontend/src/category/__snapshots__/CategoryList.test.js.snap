// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-td5piu-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-11 MuiGrid-grid-lg-10 css-x02yn7-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-exfl4s-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Business"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Business
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Comedy"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Comedy
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Education"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Education
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Drama"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Drama
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Health"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Health
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Music"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Music
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/News"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                News
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Sports"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Sports
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Tech"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Tech
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/TV"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                TV
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Film"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Film
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-1 MuiGrid-grid-lg-1 css-1fo2hm6-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1388jyf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/search/Weather"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1cwrich-MuiTypography-root"
              >
                Weather
              </p>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
